### PR TITLE
provide ability to hide default UI items

### PR DESCRIPTION
**Pull request by kckst8**

### CODE REVIEW

File: packages/modules/itwin-viewer-react/src/components/app-ui/widgets/BasicNavigationWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+/** Basic Navigation Widget that provides standard tools to manipulate views containing element data.
+ * Supports the specification of additional horizontal and vertical toolbar items through props.
+ */
+export function BasicNavigationWidget({ config }: BasicNavigationWidgetProps) {
+  const getHorizontalToolbarItems = React.useCallback((): CommonToolbarItem[] => {
+    if (config?.hideDefaultHorizontalItems) {
+      return [];
+    }
+    const items: CommonToolbarItem[] = ToolbarHelper.createToolbarItemsFromItemDefs(
```
Comment (archana-maharjan) [inline packages/modules/itwin-viewer-react/src/components/app-ui/widgets/BasicNavigationWidget.tsx:36]: Everything looks good to me, except just for curiosity, I wonder if is it okay to use widgets e.g BasicNavigationWIdget from ui-framework? I am not sure if ui-framework is open source or not, however, I remember Bill Steinbock's presentation, where he said ui-framework provides most of the widget we need and reduces all these repetition. Also, I haven't verified whether widgets provided by Ui-Framework are hideable or not, but they sure are responsive.

Comment (kckst8) [reply packages/modules/itwin-viewer-react/src/components/app-ui/widgets/BasicNavigationWidget.tsx:36]: That's what I am replacing with this one. This is basically the same widget but it adds a prop for configuration to hide the default tools. The core version does not allow this. It only allows expansion

File: packages/apps/viewer-sample-react/src/components/home/AuthClientHome.tsx [hunk 2]
```diff
@@ -79,6 +79,23 @@ export const AuthClientHome = () => {
           appInsightsKey={process.env.REACT_APP_APPLICATION_INSIGHTS_KEY}
           backend={backend}
           theme={ColorTheme.Dark}
+          defaultUiConfig={{
+            contentManipulationTools: {
+              hideDefaultHorizontalItems: true,
```
Comment (Carl-Hinkle) [inline packages/apps/viewer-sample-react/src/components/home/AuthClientHome.tsx:84]: Do we need more granularity than this?  Or is the idea that you disable the entire group and then add back things that you want?

Comment (kckst8) [reply packages/apps/viewer-sample-react/src/components/home/AuthClientHome.tsx:84]: yes. That was the idea. My thinking is to that the general goal is to use the UI Providers in UI2.0 to build the toolbars that each application desires. This basically would give them a clean slate. We can expand it into individual buttons if we see a large base that desires that, but it's just as easy (and preferable) to add the core tools to their own UI providers.

File: packages/modules/itwin-viewer-react/src/components/app-ui/widgets/BasicNavigationWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    return items;
+  }, [config]);
+
+  const getVerticalToolbarItems = React.useCallback((): CommonToolbarItem[] => {
+    if (config?.hideDefaultVerticalItems) {
+      return [];
+    }
+    const items: CommonToolbarItem[] = [];
+    items.push(
+      ToolbarHelper.createToolbarItemFromItemDef(10, CoreTools.walkViewCommand),
```
Comment (Carl-Hinkle) [inline packages/modules/itwin-viewer-react/src/components/app-ui/widgets/BasicNavigationWidget.tsx:55]: Is the LookAndMove tool available? If so we should replace the walkViewCommand with that.

**Pull request merged**